src/NUnitCore/core/Compatibility.cs

```diff
@@ -0,0 +1,298 @@
+                        bool ignore = (bool)Reflect.GetPropertyValue(attribute, "Ignore");
+                        if (ignore)
+                            Error(location, "TestFixtureAttribute Ignore property has a string value in NUnit 3. Fix after conversion.");
+                        break;
+                    case "NUnit.Framework.TestFixtureSetUpAttribute":
+                        Error(location, "TestFixtureSetUpAttribute is not supported in NUnit 3. Use OneTimeSetUpAttribute.");
+                        break;
+                    case "NUnit.Framework.TestFixtureTearDownAttribute":
+                        Error(location, "TestFixtureTearDownAttribute is not supported in NUnit 3. Use OneTimeTearDownAttribute.");
+                        break;
+                    case "NUnit.Framework.ValueSourceAttribute":
+                        sourceName = (string)Reflect.GetPropertyValue(attribute, "SourceName");
+                        if (!string.IsNullOrEmpty(sourceName))
+                        {
+                            Type sourceType = (Type)Reflect.GetPropertyValue(attribute, "SourceType");
+                            if (sourceType == null && parameterInfo != null)
+                                sourceType = parameterInfo.Member.ReflectedType;
+                            if (sourceType != null)
+                            {
+                                var members = sourceType.GetMember(sourceName, ALL_MEMBERS);
+                                if (members.Length > 0 && !IsStaticMember(members[0]))
+                                    Error(sourceType + "." + members[0].Name, "ValueSourceAttribute must reference a static member.");
+                            }
+                        }
+                        break;
+
+                    case "System.STAThreadAttribute":
+                    case "System.MTAThreadAttribute":
+                        Warning(location, attributeFullName + " has no effect in NUnit 3. Use ApartmentAttribute.");
+                        break;
+                }
+            }
+        }
+
+        #endregion
+
+        #region Helper Methods
+
+        private static bool IsStaticMember(MemberInfo member)
+        {
+            switch (member.MemberType)
+            {
+                case MemberTypes.Field:
+                    return ((FieldInfo)member).IsStatic;
+                case MemberTypes.Property:
+                    var getMethod = ((PropertyInfo)member).GetGetMethod();
+                    return getMethod != null && getMethod.IsStatic;
+                case MemberTypes.Method:
+                    return ((MethodInfo)member).IsStatic;
+                default:
+                    return false;
+            }
+        }
+
+        private static void Report(string level, string location, string message)
+        {
+            if (_writer != null)
+                _writer.WriteLine(string.Format(REPORT_FORMAT, level, location, message));
```

Comment:
Oh, I see why it's important to null out `_writer` now.

Reply:
Only an option if we go back to the future. 😄 We're in the past here!